tools/federation/src/main/java/org/eclipse/rdf4j/federated/evaluation/concurrent/ControlledWorkerScheduler.java

```diff
@@ -46,23 +46,41 @@
 
 	private final int nWorkers;
 	private final String name;
+	private final TaskWrapper taskWrapper;
 
 	/**
 	 * Construct a new instance with 20 workers.
+	 * 
+	 * @deprecated use {@link #ControlledWorkerScheduler(int, String, TaskWrapper)}. Scheduled to be removed in 4.0
 	 */
+	@Deprecated
 	public ControlledWorkerScheduler() {
-		this(20, "FedX Worker");
+		this(20, "FedX Worker", DefaultTaskWrapper.INSTANCE);
 	}
 
 	/**
 	 * Construct a new instance with the specified number of workers and the given name.
 	 *
 	 * @param nWorkers
 	 * @param name
+	 * @deprecated use {@link #ControlledWorkerScheduler(int, String, TaskWrapper)}. Scheduled to be removed in 4.0
 	 */
+	@Deprecated
 	public ControlledWorkerScheduler(int nWorkers, String name) {
+		this(nWorkers, name, DefaultTaskWrapper.INSTANCE);
+	}
+
+	/**
+	 * Construct a new instance with the specified number of workers, the given name and {@link TaskWrapper}.
+	 *
+	 * @param nWorkers
+	 * @param name
+	 * @param taskWrapper
+	 */
+	public ControlledWorkerScheduler(int nWorkers, String name, TaskWrapper taskWrapper) {
```

Comment:
as a TaskWrapper might be a construct also used in other places and there are already quite some constructor parameters and constructor variants I suggest to rather create an interface TaskWrapperAware which allows setting  (and getting) a TaskWrapper and implement that here. This also allows usiing this in other places.